pkg/common/telemetry/specific_metrics.go

```diff
@@ -0,0 +1,428 @@
+package telemetry
+
+import (
+	"strconv"
+	"time"
+)
+
+// Call Counters (timing and success metrics)
+// Allows adding labels in-code
+
+// AddAttestorType add Attestor type label to the given counter
+// from the given attestor type
+func AddAttestorType(cc *CallCounter, aType string) {
+	cc.AddLabel(Attestor, aType)
+}
+
+// AddAudience add the Audience label(s) to the given counter
+// from the given audience(s)
+func AddAudience(cc *CallCounter, auds ...string) {
+	for _, aud := range auds {
+		cc.AddLabel(Audience, aud)
+	}
+}
+
+// AddCallerID add the CallerID label to the given counter
+// from the given ID
+func AddCallerID(cc *CallCounter, id string) {
+	cc.AddLabel(CallerID, id)
+}
+
+// AddRegistered add the Registered label to the given
+// counter, from given boolean
+func AddRegistered(cc *CallCounter, reg bool) {
+	cc.AddLabel(Registered, strconv.FormatBool(reg))
+}
+
+// AddSPIFFEID add SPIFFE ID label to the given counter
+// from the given ID
+func AddSPIFFEID(cc *CallCounter, id string) {
+	cc.AddLabel(SPIFFEID, id)
+}
+
+// ManagerFetchUpdatesSyncCall return a call counter for
+// the agent's manager fetching updates on sync
+func ManagerFetchUpdatesSyncCall(m Metrics) *CallCounter {
```

Comment:
should these kind of methods be prefixed with `Start`? I know it makes them even longer but they'd be more "verby".

Reply:
`ManagerFetchUpdatesSyncCall` is particularly confusing name in isolation.
I have no idea what action it represents, and the comment "agent's manager fetching updates on sync" offers me no additional clue.

What is being fetched? What is being updated? Is it important that its an update vs a create?
Why is `sync` important? Is there an asynchronous version that is a different metric?
Why is manager more important than agent?

Seems like the name is based on _part_ of the file path and method `agent/manager/sync.go: func (m *manager) fetchUpdates` . File path and method may change and are less important than functionality represented.

`fetchUpdates` seems designed to get a new SVID for the agent, and update the cache. Is it important for telemetry to denote cache operations?

How about `FetchAgentSVIDCall`?

How is this different from `NodeAPIFetchX509SVIDCall`?